torch_xla/csrc/pooling.cpp

```diff
@@ -144,7 +144,23 @@ std::vector<std::pair<int64_t, int64_t>> CeilModePadding(
         (input_size + 2 * left_padding - kernel_size[i]) % stride[i];
     int64_t right_padding = left_padding;
     if (ceil_mode && output_size_rem != 0) {
-      right_padding += stride[i];
+      int64_t extra_padding = stride[i] - output_size_rem;
+      int64_t new_output_size =
+          (input_size + left_padding + right_padding + extra_padding -
+           kernel_size[i] + stride[i] - 1) /
+              stride[i] +
+          1;
```

Comment:
lol this computation is really puzzling. reminds me other leetcode questions.

Reply:
this is what was there before: https://github.com/pytorch/xla/pull/6409/files#diff-b7dcec47bb003c09d0f32235e139ca6f1bb411e30a5194a68d9745cb25032371L143

I thought those are too complex and useless so i removed them... turns out it's useful something